src/api/tigl.cpp

```diff
@@ -4813,6 +4916,106 @@ TIGL_COMMON_EXPORT TiglReturnCode tiglIntersectGetPoint(TiglCPACSConfigurationHa
+    }
+    if (!pointY) {
+        LOG(ERROR) << "Null pointer for argument pointY in tiglGetCurveIntersectionPoint.";
+        return TIGL_NULL_POINTER;
+    }
+    if (!pointZ) {
+        LOG(ERROR) << "Null pointer for argument pointZ in tiglGetCurveIntersectionPoint.";
+        return TIGL_NULL_POINTER;
+    }
+    if (!intersectionID) {
+        LOG(ERROR) << "Null pointer for argument intersectionID in tiglGetCurveIntersectionPoint.";
+        return TIGL_NULL_POINTER;
+    }
+
+    try {
+        tigl::CCPACSConfigurationManager& manager = tigl::CCPACSConfigurationManager::GetInstance();
+        tigl::CCPACSConfiguration& config = manager.GetConfiguration(cpacsHandle);
+        tigl::CTiglShapeCache& cache = config.GetShapeCache();
+
+        tigl::CTiglIntersectionCalculation Intersector(cache, intersectionID);
+
+        int npoints = Intersector.GetCountIntersectionPoints();
+        if ( pointIdx < 1 || pointIdx > npoints ) {
+            return TIGL_INDEX_ERROR;
+        }
+
+        gp_Pnt p = BRep_Tool::Pnt( Intersector.GetVertex(pointIdx) );
+
+        *pointX = p.X();
+        *pointY = p.Y();
+        *pointZ = p.Z();
+
+        return TIGL_SUCCESS;
+    }
+    catch (const tigl::CTiglError& ex) {
+        LOG(ERROR) << ex.what();
+        return ex.getCode();
+    }
+    catch (...) {
+        LOG(ERROR) << "Caught an exception in tiglGetCurveIntersectionPoint!";
+        return TIGL_ERROR;
+    }
+}
+
+TIGL_COMMON_EXPORT TiglReturnCode tiglGetCurveIntersectionParameter (TiglCPACSConfigurationHandle cpacsHandle,
+                                                                     const char* intersectionID,
+                                                                     int pointIdx,
+                                                                     const char* curveID,
+                                                                     int curveIdx,
+                                                                     double* eta)
+{
+    if (!eta) {
+        LOG(ERROR) << "Null pointer for argument eta in tiglGetCurveIntersectionParameter .";
+        return TIGL_NULL_POINTER;
+    }
+    if (!intersectionID) {
+        LOG(ERROR) << "Null pointer for argument intersectionID in tiglGetCurveIntersectionParameter .";
+        return TIGL_NULL_POINTER;
+    }
+    if (!curveID) {
+        LOG(ERROR) << "Null pointer for argument curveID in tiglGetCurveIntersectionParameter .";
+        return TIGL_NULL_POINTER;
+    }
+
+    try {
+        tigl::CCPACSConfigurationManager& manager = tigl::CCPACSConfigurationManager::GetInstance();
+        tigl::CCPACSConfiguration& config = manager.GetConfiguration(cpacsHandle);
+        tigl::CTiglShapeCache& cache = config.GetShapeCache();
+
+        tigl::CTiglIntersectionCalculation IntersectorPoint(cache, intersectionID);
+        gp_Pnt p = BRep_Tool::Pnt( IntersectorPoint.GetVertex(pointIdx) );
+
+        tigl::CTiglIntersectionCalculation IntersectorLine(cache, curveID);
```

Comment:
What happens, it the user chooses '''curveID''' to be a curve that was not part of the intersection '''intersectionID'''?

This is principally possible. In this case, the result makes no sense.

Would it make sense to decouple the functionality into:
 - >GetIntersectionPoint(curveId, curveIdx, pointIdx) -> PointCoordinates
 -> GetCurveParameter(curveId, curveIdx, pointCoordinates) -> PointParameter, ProjectionError

The GetCurveParameter would then be usable more generally.